module/zfs/dmu_objset.c

```diff
@@ -796,6 +797,9 @@ dmu_objset_create_impl(spa_t *spa, dsl_dataset_t *ds, blkptr_t *bp,
 	os->os_phys->os_type = type;
 	if (dmu_objset_userused_enabled(os)) {
 		os->os_phys->os_flags |= OBJSET_FLAG_USERACCOUNTING_COMPLETE;
+		if (dmu_objset_userobjused_enabled(os))
+			os->os_phys->os_flags |=
+			    OBJSET_FLAG_USEROBJACCOUNTING_COMPLETE;
```

Comment:
So I was confused about this, see my [comment](https://github.com/zfsonlinux/zfs/pull/3983#issuecomment-223637670) below.  We can safely assume all pools have user/group space accounting.